core/src/services/supabase/backend.rs

```diff
@@ -259,16 +286,38 @@ impl Accessor for SupabaseBackend {
             return Ok(RpStat::new(Metadata::new(EntryMode::DIR)));
         }
 
-        let resp = self.core.supabase_get_object_info(path).await?;
+        // The get_object_info does not contain the file size. Therefore
```

Comment:
Why? object should contain the metadata of size.

Reply:
The reason here is that the supabase stores their actual response in the body.
For example, if status_core is 404, it is in the body, not in the header (which is really weird....).

Hence here if we want to get the metadata like filesize, we have to use `HEAD`. And if any error returns, we send another HTTP request to `/object/info` to actually get the error.

Reply:
I didn't get it. If status code is OK, the response is not object metadata?

Reply:
> I didn't get it. If status code is OK, the response is not object metadata?

If we use `supabase_get_object_info()`, the content_length in the header of the response will not be the file size (actually the response has nothing to do with the file size). But this behavior is required when we use `stat`.

Reply:
Hi, here is a misunderstanding. `stat`'s implementation is nothing related to HEAD request nor Content Length header. Please take gcs's stat as a reference.

Reply:
> I didn't get it. If status code is OK, the response is not object metadata?

I think the object info is actually returned in the body, but I cannot find the schema in its documentation.

Reply:
> but I cannot find the schema in its documentation.

I believe it's another new issue😂. For now, we can parse it from the real output.

Reply:
> I believe it's another new issue😂. For now, we can parse it from the real output.

Another weird thing. I tried to print the body and get a handmade schema. Here is the code I used.
```Rust
        let resp = self.core.supabase_get_object_info(path).await?;

        let (header, body) = resp.into_parts();
        println!("{:?}", &header);
        println!("{:?}", &body.bytes().await?);
```

This is what is printed
```text
Parts { status: 200, version: HTTP/1.1, headers: {"content-type": "application/octet-stream", 
"content-length": "0", "connection": "keep-alive", "accept-ranges": "bytes", 
"etag": "\"d7f191a1a4e3395eb8cd372561486268\"", "last-modified": "Wed, 03 May 2023 13:44:46 GMT", 
"cache-control": "no-cache", "date": "Wed, 03 May 2023 13:44:46 GMT", 
"access-control-allow-origin": "*", "x-kong-upstream-latency": "30", 
"x-kong-proxy-latency": "0", "via": "kong/2.8.1"} }
b""
```

which means the body is empty, and the content-length is zero 🤣 . Seems there are no workarounds other than doing what is done in the PR.

Reply:
supabase, It's not easy to say I love you...

Reply:
Ok, we don't need to call `supabase_get_object_info` for error response, just transform error from http status code is enough. We will contact with supabase team for better solutions.

Reply:
> which means the body is empty, and the content-length is zero

I believe it's bug, supabase storage should:

- Either, Return an json with metadata in body for `GET /object/info`
- Or, use `HEAD /object/info` to return metadata which can have correct content length.
  - Maybe conflict with `HEAD /object`

Reply:
> Ok, we don't need to call `supabase_get_object_info` for error response, just transform error from http status code is enough. We will contact with supabase team for better solutions.
> 
> storage-api has a special `head` render for this: https://github.com/supabase/storage-api/blob/1794d04ab0e7259d9058293e257f442015252ec3/src/storage/renderer/head.ts#L6-L23
> 
> More like a bug...

AFAIK, supabase always return error 400 on all failures, and the actual status_code is in the body...
Like this
```text
Parts { status: 400, version: HTTP/1.1, headers: {"content-type": "application/json; charset=utf-8", "content-length": "69", "connection": "keep-alive", "date": "Wed, 03 May 2023 16:45:27 GMT", "access-control-allow-origin": "*", "x-kong-upstream-latency": "41", "x-kong-proxy-latency": "5", "via": "kong/2.8.1"} }

b"{\"statusCode\":\"404\",\"error\":\"not_found\",\"message\":\"Object not found\"}"
```

Reply:
> I believe it's bug, supabase storage should:

I agree, the current storage RESTful API seems not being carefully designed.

Reply:
> AFAIK, supabase always return error 400 on all failures, and the actual status_code is in the body...

Ohhhh😅

Reply:
Can you help report this use case to supabase too? Thanks.